src/main/scala/viper/silver/reporter/Reporter.scala

```diff
@@ -20,13 +23,38 @@ object NoopReporter extends Reporter {
   def report(msg: Message): Unit = ()
 }
 
-case class CSVReporter(name: String = "csv_reporter", path: String = "report.csv") extends Reporter {
+trait PluginAwareReporter extends Reporter {
+
+  var plugins: Option[SilverPluginManager] = None
+
+  def setPluginManager(pm: Option[SilverPluginManager]): Unit = {
+    plugins = pm
+  }
+
+  override def report(msg: Message): Unit = {
```

Comment:
Should we make this function `final` to avoid accidentally overriding this function?
```suggestion
  final override def report(msg: Message): Unit = {
```